### PR TITLE
[libc] Fix missing sysroot path for kernel headers when crosscompiling

### PR DESCRIPTION
**Pull request by mikhailramalho**
When crosscompiling, we need to search for the linux kernel headers in the sysroot but since #97486 the linux kernel headers were always searched in /usr/include.

This patch fixes this behaviour by prepending a '=' to where we search for the kernel headers. As per the gcc/clang's documentation a '=' before the path is replaced by the sysroot.

### CODE REVIEW

File: libc/src/time/linux/nanosleep.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/time/nanosleep.h"
-
+#include "hdr/time_macros.h"
```
Comment (petrhosek) [inline libc/src/time/linux/nanosleep.cpp:10]: This change seems unrelated, can you move it to a separate PR?

**Review by lntue — APPROVED** [review]

**Comment by nickdesaulniers** [comment]
So when cross compiling, we should only look at the sysroot, and not /usr/include, right?

**Comment by mikhailramalho** [comment]
> So when cross compiling, we should only look at the sysroot, and not /usr/include, right?

IIUC, we need to add the `$sysroot/usr/include` explicitly because we add `-nostdlibinc` by default.
